lib/manageiq/api/common/application_controller_mixins/exception_handling.rb

```diff
@@ -0,0 +1,43 @@
+module ManageIQ
+  module API
+    module Common
+      module ApplicationControllerMixins
+        module ExceptionHandling
+          DEFAULT_ERROR_CODE = 400
+
+          def self.included(other)
+            other.rescue_from(StandardError, RuntimeError) do |exception|
+              errors, status = collected_errors(exception)
+
+              render :json => errors, :status => status
+            end
+          end
+
+          def collected_errors(exception)
+            errors = []
+            top_level_exception = exception
+
+            until exception.nil?
+              code = exception.respond_to?(:code) ? exception.code : exception.class.to_s
+              errors << {:status => code, :detail => exception.class.to_s + ": " + exception.message}
```

Comment:
Wouldn't this wind up setting `{:status => "SomeClass", ...}` if the exception doesn't respond to `#code`?

Reply:
Yeah thats right, I'll use the error_code_from_class instead of just the exception name.

Reply:
Prefer interpolation over `+` for string building